modules/gapi/src/backends/fluid/gfluidimgproc.cpp

```diff
@@ -2088,32 +2207,44 @@ GAPI_FLUID_KERNEL(GFluidResize, cv::gapi::imgproc::GResize, true)
        }
        cv::Size outSize(outSz_w, outSz_h);
 
-       if (in.chan == 3)
+       if (in.depth == CV_8U && in.chan == 3)
        {
-           initScratchLinear<uchar, Mapper, 3>(in, outSize, scratch, LPI);
+           initScratchLinear<uchar, linear::Mapper, 3>(in, outSize, scratch, LPI);
        }
-       else if (in.chan == 4)
+       else if (in.depth == CV_32F && in.chan == 1)
        {
-           initScratchLinear<uchar, Mapper, 4>(in, outSize, scratch, LPI);
+           initScratchLinear<float, linear32f::Mapper, 1>(in, outSize, scratch, LPI);
        }
-    }
+       else
+       {
+           CV_Error(cv::Error::StsBadArg, "unsupported combination of type and number of channel");
+       }
+   }
 
     static void resetScratch(cv::gapi::fluid::Buffer& /*scratch*/)
     {}
 
-    static void run(const cv::gapi::fluid::View& in, cv::Size /*sz*/, double /*fx*/, double /*fy*/, int interp,
-                    cv::gapi::fluid::Buffer& out,
-                    cv::gapi::fluid::Buffer& scratch) {
+    static void run(const cv::gapi::fluid::View& in, cv::Size /*sz*/, double /*fx*/,
+                    double /*fy*/, int interp, cv::gapi::fluid::Buffer& out,
+                    cv::gapi::fluid::Buffer& scratch)
+    {
+        GAPI_Assert((in.meta().depth == CV_8U && in.meta().chan == 3) ||
+                    (in.meta().depth == CV_32F && in.meta().chan == 1));
+        GAPI_Assert(interp == cv::INTER_LINEAR);
         const int channels = in.meta().chan;
-        GAPI_Assert((channels == 3 || channels == 4) && (interp == cv::INTER_LINEAR));
+        const int depth = in.meta().depth;
 
-        if (channels == 3)
+        if (depth == CV_8U && channels == 3)
+        {
+            calcRowLinearC<uint8_t, linear::Mapper, 3>(in, out, scratch);
+        }
+        else if (depth == CV_32F && channels == 1)
         {
-            calcRowLinearC<uint8_t, Mapper, 3>(in, out, scratch);
+            calcRowLinear<linear32f::Mapper>(in, out, scratch);
         }
-        else if (channels == 4)
+        else
         {
-            calcRowLinearC<uint8_t, Mapper, 4>(in, out, scratch);
+            CV_Error(cv::Error::StsBadArg, "unsupported combination of type and number of channel");
```

Comment:
Is there a way to report what actually combination failed in troubleshooting purpoese?